pkg/lib/interaction/nodes/create_authenticator_magic_link_otp_setup.go

```diff
@@ -0,0 +1,141 @@
+package nodes
+
+import (
+	"github.com/authgear/authgear-server/pkg/api/model"
+	"github.com/authgear/authgear-server/pkg/lib/authn"
+	"github.com/authgear/authgear-server/pkg/lib/authn/authenticator"
+	"github.com/authgear/authgear-server/pkg/lib/authn/otp"
+	"github.com/authgear/authgear-server/pkg/lib/feature/verification"
+	"github.com/authgear/authgear-server/pkg/lib/interaction"
+)
+
+func init() {
+	interaction.RegisterNode(&NodeCreateAuthenticatorMagicLinkOTPSetup{})
+}
+
+type InputCreateAuthenticatorMagicLinkOTPSetup interface {
+	GetMagicLinkOTPTarget() string
+}
+
+type EdgeCreateAuthenticatorMagicLinkOTPSetup struct {
+	NewAuthenticatorID string
+	Stage              authn.AuthenticationStage
+	IsDefault          bool
+}
+
+type InputCreateAuthenticatorMagicLinkOTPSetupSelect interface {
+	SetupPrimaryAuthenticatorMagicLinkOTP()
+}
+
+func (e *EdgeCreateAuthenticatorMagicLinkOTPSetup) IsDefaultAuthenticator() bool {
+	return false
+}
+
+func (e *EdgeCreateAuthenticatorMagicLinkOTPSetup) AuthenticatorType() model.AuthenticatorType {
+	// Currently only support send through email
+	return model.AuthenticatorTypeOOBEmail
+}
+
+func (e *EdgeCreateAuthenticatorMagicLinkOTPSetup) Instantiate(ctx *interaction.Context, graph *interaction.Graph, rawInput interface{}) (interaction.Node, error) {
+	var userID string
+	var input InputCreateAuthenticatorMagicLinkOTPSetup
+	if e.Stage == authn.AuthenticationStagePrimary {
+		panic("Magic link as primary authenticator is not yet supported")
+	} else {
+		if !interaction.Input(rawInput, &input) {
+			return nil, interaction.ErrIncompatibleInput
+		}
+		userID = graph.MustGetUserID()
+	}
+	spec := &authenticator.Spec{
+		UserID:    userID,
+		IsDefault: e.IsDefault,
+		Kind:      stageToAuthenticatorKind(e.Stage),
+		Type:      e.AuthenticatorType(),
+		OOBOTP: &authenticator.OOBOTPSpec{
+			Email: input.GetMagicLinkOTPTarget(),
+		},
+		MagicLinkOTP: &authenticator.MagicLinkOTPSpec{
+			Email: input.GetMagicLinkOTPTarget(),
+		},
+	}
+
+	info, err := ctx.Authenticators.NewWithAuthenticatorID(e.NewAuthenticatorID, spec)
+	if err != nil {
+		return nil, err
+	}
+
+	var skipInput interface{ SkipVerification() bool }
+	if interaction.Input(rawInput, &skipInput) && skipInput.SkipVerification() {
+		// Admin skip verify MagicLink otp and create OOB authenticator directly
+		return &NodeCreateAuthenticatorMagicLinkOTP{Stage: e.Stage, Authenticator: info}, nil
+	}
+
+	aStatus, err := ctx.Verification.GetAuthenticatorVerificationStatus(info)
+	if err != nil {
+		return nil, err
+	}
+
+	if aStatus == verification.AuthenticatorStatusVerified {
+		return &NodeCreateAuthenticatorOOB{Stage: e.Stage, Authenticator: info}, nil
```

Comment:
Given that `NodeCreateAuthenticatorMagicLinkOTPSetup` is pretty different from other OTP flows, I think it would be better to separate other nodes e.g in this case `NodeCreateAuthenticatorMagicLinkOTP` as well?